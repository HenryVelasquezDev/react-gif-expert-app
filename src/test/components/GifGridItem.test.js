import React from 'react';
//import { render } from '@testing-library/react'; 
import GifGridItem from '../../components/GifGridItem';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

describe('Pruebas en <GifGridItem />', () => {
    
    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg'; 
    const wrapper = shallow( <GifGridItem title ={title} url={url}  /> );
    
    //pruebas despues de instalar enzyme en setupTests.js
    test('debe mostrar <GifGridItem /> correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe tener un parrafo con el title ', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    });

    test('debe de obtener la imagen igual al url y alt de los props', () => {
       
        const img = wrapper.find( 'img' );
        //console.log( img.prop('src') );
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });
    
    test('debe de tener animate__bounceInLeft', () => {
        const div = wrapper.find( 'div' );
        const className = div.prop('className');
        //console.log(div.prop('className').split(' '));
        expect( className.split(' ')[2] ).toBe( 'animate__bounceInLeft' );
        expect( className.includes('animate__bounceInLeft') ).toBe ( true );
    })
    

})
