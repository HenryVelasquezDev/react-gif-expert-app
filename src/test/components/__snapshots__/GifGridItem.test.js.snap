// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__bounceInLeft"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;
